src/stack.js

```diff
@@ -1,5 +1,49 @@
 'use strict'
 
+class Node {
+  constructor(data, next) {
+    this.data = data
+    this.next = next
+  }
+}
+
 export default class Stack {
-  // your code here
+  constructor() {
+    this.top = 0
+    this.size = 0
+  }
+
+  push(input) {
+    this.size++
+    this.top = new Node(input, this.top)
+  }
+
+  pop() {
+    if(this.size === 0) {
+      return null
+    }
+    this.size--
+    let poppedItem = this.top.data
+    this.top = this.top.next
+    return poppedItem
+  }
+
+  peek() {
+    if (this.size === 0) {
+      return null
+    }
+    return this.top.data
+  }
+
+  isEmpty() {
```

Comment:
This could also be 
```javascript
return this.size === 0
```
or if you prefer ternary operators
```javascript
return this.size
  ? true
  : false
```